histomicstk/utils/linalg.py

```diff
@@ -0,0 +1,16 @@
+import numpy, numpy.linalg
+
+def get_principal_components(m):
+    """Take a matrix m (probably 3xN) and return the 3x3 matrix of
+    "principal components".  (Actually computed with SVD)
+
+    """
+    return numpy.linalg.svd(m.astype(float), full_matrices=False)[0].astype(m.dtype)
+
+def magnitude(m):
+    """Get the magnitude of each column vector in a matrix"""
+    return numpy.sqrt((m ** 2).sum(0))
+
+def normalize(m):
+    """Normalize each column vector in a matrix"""
+    return m / magnitude(m)
```

Comment:
`normalize` in particular is, with these changes, used in `color_deconvolution` (the routine), `macenko_stain_matrix`, and `snmf_stain_matrix`. I can move these elsewhere, but at least `normalize` and thus `magnitude` should be usable across those modules, at least.